### ISSUE TITLE
Wherigo: Missing showing the new image of the item

### ISSUE THREAD

**Issue by rouroni1** [cgeo/cgeo#16170]
### Describe your problem!

When playing wherigo from GCAC95E, on entering the item Boat, the cartridge is programmed to change the item image from a Empty Boat to a Boat wit a Sailor, but on C:Geo goes directly to the options menu.

Inputs :
Shows on top "Input Dialog" and shouldn't show.

Dialogs
Shows on top "Dialog" and shouldn't show.
Have "Cancel", and shouldn't. The cancel button does the same than Ok button.

Zones:
Locate on center button? (For testing???)

https://github.com/user-attachments/assets/347cb4a0-6acf-44f9-832d-f3ca8de36f56

### How to reproduce?

1. Start Cartridge
2. Select language (Input)
3. Accept place rules (Input)
4. Enter Right Bank (Zone)
5. Select item Boat and Enter the Boat.

### Actual result after these steps?

The game goes directly to the options menu.

### Expected result after these steps?

Should Show the new image of the boat, and the button Ok

### Reproducible

Yes

### c:geo Version

2024.10.05-NB-33b5ab3

### System information

```text
## System information

c:geo version: 2024.10.05-NB-33b5ab3

Device:
-------
- Device type: 21121210G (ingres_eea, POCO)
- Available processors: 8
- Android version: 14
- Android build: UKQ1.230917.001 release-keys
- Screen resolution: 1080x2186px (393x795dp)
- Pixel density: 2.75
- System font scale: 1.0 / used scale: 1.0
- Sailfish OS detected: false
- Google Play services: enabled - 24.37.34 (190400-679256246)
- Memory: Available:2,9 GB, Total:10,9 GB, Threshold: 216,0 MB, low:false

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: orientation

Program settings:
-------
- Settings: v9, Count:203
- Set language: pt_PT (system default)
- System date format: dd/MM/yy
- Time zone: GMT+01:00
- Debug mode active: no
- Log Settings: debug=false, minLevel=WARN, minAddCallerInfo=NONE, addCallerInfoMaxDepth=8, throwOnError=false, transactionSizes=false
- Last manual backup: 3/10, 23:05
- Last auto backup: 28/09, 08:40
- Routing mode: Walk (shortest.brf / trekking.brf / car-eco.brf / (none) / (none))
- Live map mode: true
- Use unified map: true
- OSM multi-threading: false / threads: 1
- Map: Spain-Portugal (OAM) (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMapSource:primary:CGEO/maps/Spain-Portugal (OAM).map
  - Atts: #VectorMaps:(c)Christian Kernbeis, GeneralMaps:(c)Michael Bechtold, Themes:(c)Tobias Kuehn, www.openandromaps.org 2012-2024 #Hoehendaten: sonny.4lima.de, viewfinderpanoamas.org, NASA #Kartendaten: openstreetmap.org
  - Theme: Elevate.zip:Elevate.xml

Filters:
-------
- Hide waypoints: original 
- LIVE: Estado: Minhas=Não, Tem Registo Offline de Encontrada=Não ({"advanced":false,"inconclusive":false,"tree":{"type":"AND","children":[{"type":"type","config":{}},{"type":"difficulty_terrain","config":{"d":{},"t":{}}},{"type":"status","config":{"values":["owned_no","has_offline_found_log_no"]}}]}})
- OFFLINE: Estado: 4 itens ({"advanced":true,"inconclusive":false,"tree":{"type":"AND","children":[{"type":"type","config":{}},{"type":"status","config":{"values":["owned_no","found_no","has_offline_log_no","has_offline_found_log_no"]}}]}})

Services:
-------
- Geocaching sites enabled:
   - geocaching.com: Logged in (Sessão iniciada) / PREMIUM
   - Geocaching.com Adventure Lab
- Geocaching.com date format: yyyy-MM-dd
- Geocaching.com website language: pt-PT
- Last login error on geocaching.com: communication error // (2024-10-05 09-52)
- Last successful login on geocaching.com: 2024-10-05 17-47
- Routing: internal / BRouter installed: false

Permissions
-------
- ACCESS_COARSE_LOCATION: granted
- ACCESS_FINE_LOCATION: granted
- READ_CONTACTS: granted
- WRITE_EXTERNAL_STORAGE: DENIED
- POST_NOTIFICATIONS: granted

Paths
-------
- System internal c:geo dir: /data/user/0/cgeo.geocaching (148,6 GB free) v3 internal isDir(10 entries)
- Legacy User storage c:geo dir: /data/user/0/cgeo.geocaching (148,6 GB free) v3 internal isDir(10 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (148,6 GB free) v3 external non-removable isDir(7102 entries)
- Internal theme sync (is turned ON): /data/user/0/cgeo.geocaching/MapThemeData (148,6 GB free) v3 internal isDir(9 entries)
- Map render theme path: Elevate.zip:Elevate.xml
- Public Folders: #13
  - BASE: content://com.android.externalstorage.documents/tree/primary:CGEO/document/primary:CGEO (av:true, files:>=23, size:>=1,9 GB, free:>=148,6 GB)
  - OFFLINE_MAPS: content://com.android.externalstorage.documents/tree/primary:CGEO/maps/document/primary:CGEO/maps (av:true, files:>=10, size:>=1,9 GB, free:>=148,6 GB)
  - OFFLINE_MAP_THEMES: content://com.android.externalstorage.documents/tree/primary:CGEO/maps/_themes/document/primary:CGEO/maps/_themes (av:true, files:>=8, size:>=1,9 MB, free:>=148,6 GB)
  - OFFLINE_MAP_SHADING: content://com.android.externalstorage.documents/tree/primary:CGEO/maps/_hgt/document/primary:CGEO/maps/_hgt (av:true, files:>=0, size:>=0 B, free:>=148,6 GB)
  - LOGFILES: content://com.android.externalstorage.documents/tree/primary:CGEO/document/primary:CGEO/logfiles (av:true, files:>=12, size:>=501,0 KB, free:>=148,6 GB)
  - GPX: content://com.android.externalstorage.documents/tree/primary:CGEO/document/primary:CGEO/gpx (av:true, files:>=1, size:>=226,1 KB, free:>=148,6 GB)
  - BACKUP: content://com.android.externalstorage.documents/tree/primary:CGEO/document/primary:CGEO/backup (av:true, files:>=14, size:>=600,9 MB, free:>=148,6 GB)
  - FIELD_NOTES: content://com.android.externalstorage.documents/tree/primary:CGEO/document/primary:CGEO/field-notes (av:true, files:>=0, size:>=0 B, free:>=148,6 GB)
  - SPOILER_IMAGES: content://com.android.externalstorage.documents/tree/primary:CGEO/document/primary:CGEO/GeocachePhotos (av:true, files:>=0, size:>=0 B, free:>=148,6 GB)
  - ROUTING_BASE: content://com.android.externalstorage.documents/tree/primary:CGEO/document/primary:CGEO/routing (av:true, files:>=11, size:>=184,8 MB, free:>=148,6 GB)
  - ROUTING_TILES: content://com.android.externalstorage.documents/tree/primary:CGEO/document/primary:CGEO/routing/segments4 (av:true, files:>=3, size:>=184,7 MB, free:>=148,6 GB)
  - WHERIGO: content://com.android.externalstorage.documents/tree/primary:CGEO/document/primary:CGEO/wherigo (av:true, files:>=3, size:>=4,9 MB, free:>=148,6 GB)
  - TEST_FOLDER: file:///data/user/0/cgeo.geocaching/files/unittest (av:true, files:>=0, size:>=0 B, free:>=148,6 GB)
- PersistedDocumentUris: #2
  - PROXIMITY_NOTIFICATION_FAR: null
  - PROXIMITY_NOTIFICATION_CLOSE: null
- Persisted Uri Permissions: #4
  - content://com.android.externalstorage.documents/tree/primary:CGEO/maps/_hgt (30/01/2023, 21:22):RW
  - content://com.android.externalstorage.documents/tree/primary:CGEO/maps (30/01/2023, 21:22):RW
  - content://com.android.externalstorage.documents/tree/primary:CGEO/maps/_themes (30/01/2023, 21:22):RW
  - content://com.android.externalstorage.documents/tree/primary:CGEO (12/01/2023, 21:47):RW

Database
-------
- File: /data/user/0/cgeo.geocaching/databases/data (v103, Size:284,6 MB) on system internal storage
- Data: {cg_attributes=52983, cg_caches=7545, cg_caches_lists=8103, cg_categories=0, cg_extension=72, cg_filters=0, cg_lists=10, cg_logCount=41695, cg_logImages=146420, cg_logs=285540, cg_logs_offline=24, cg_logs_offline_images=3, cg_logs_offline_trackables=35, cg_route=0, cg_search_destination_history=0, cg_spoilers=10598, cg_trackables=407, cg_trail_history=700, cg_variables=161, cg_waypoints=5445}
- Extension Data: {DBEXTENSION_DOWNGRADEABLE_DBVERSION=1, DBEXTENSION_EMOJILRU=4, DBEXTENSION_FOUNDNUM=1, DBEXTENSION_LAST_TRACKABLE_ACTION=52, DBEXTENSION_ONE_TIME_DIALOGS=8, DBEXTENSION_POCKETQUERY_HISTORY=1, DBEXTENSION_TRACKFILES=5}

--- End of system information ---
```

### Additional Information

If needed for testing I can provide the cartridge without emulation protection

**Comment by eddiemuc** [comment]
@rouroni1 thanks a lot for this detailed report! 

I have a feeling about what is causing your main reported bug. I try to work on this as soon as possible.

Cartridge w/o protection would be nice, can you maybe send it to support@cgeo.org?

**Comment by eddiemuc** [comment]
@rouroni1 update: cartridge w/o protection is not needed, testing works well with the "productive" one

**Comment by eddiemuc** [comment]
@rouroni1 PR #16178 should fix all mentioned things with one exception:

You mention above:
- Inputs : Shows on top "Input Dialog" and shouldn't show.
- Dialogs: Shows on top "Dialog" and shouldn't show.

Question: what should I use as a title instead for Wherigo input and push dialogs? (I changed to "Wherigo Dialog" / "Wherigo Input Dialog" for now)

**Comment by eddiemuc** [comment]
Nice cartridge btw

**Comment by rouroni1** [comment]
Thank you. It was a project with some years, and made this summer.

I don't think that we need to identify everything, unless it's necessary for development.

When playing a Wherigo cartridge, ou a PC game, as a player, when I arrive to a place where I need to provide a  writen anwser or select a option to continue, the identification of that place for the player isn't neaded.

I recognize, however, that this identification is useful for those who want to know how the cartridge is/was built or for those developing the application to play the cartridge. But only for that, for the player, it is unnecessary.

I attached a PDF with the images captured in cgeo and whereyougo, and the notes beside it indicating my opinion.

[GCAC95E.pdf](https://github.com/user-attachments/files/17269119/GCAC95E.pdf)

**Comment by eddiemuc** [comment]
Thanks for the detailled analysis. All your comments should be fixed with PR #16178
